services/document_service.go

```diff
@@ -1,18 +1,37 @@
 package services
 
 import (
-	"time"
-
 	"github.com/docentre/docentre/models"
 	"github.com/docentre/docentre/repositories"
 )
 
 func CreateDocument(authorID uint) (models.Document, error) {
-	doc := models.Document{AuthorID: authorID, LastEditDate: time.Now()}
-	result := repositories.DB.Create(&doc)
+	doc := models.Document{AuthorID: authorID}
+	result := repositories.DB.Omit("ApprovedDate, ApproverID, ViewerID").Create(&doc)
 	if result.Error != nil {
 		return models.Document{}, result.Error
 	}
 
 	return doc, nil
 }
+
+func UpdateDocument(documentID uint, authorID uint, title string, content string, appendix string, status string, approverID uint) (uint, error) {
+
+	if approverID == 0 {
+		result := repositories.DB.Model(&models.Document{}).Where("id = ?", documentID).Updates(map[string]interface{}{"author_id": authorID, "title": title, "content": content, "appendix": appendix, "status": status})
```

Comment:
What will happen to the `approver_id` in the database if it's not set? How about setting it as `0` as well? This would make the if-branch and else-branch identical, allowing them to be merged.

Reply:
set to 0 is not allow since it is a FK to User.ID

Reply:
Understood. How about we move the duplicate code outside of the condition? As you can see, the only difference is the `map[string]interface{}{...}`. We could assign this to a variable in the condition and execute the update and error checking outside of the condition.